### ISSUE TITLE
Very slow resolution for multimodule project.

### ISSUE THREAD

**Issue by jilen** [coursier/coursier#1272]
sbt: 1.2.8
sbt-coursier: 2.0.0-RC2
Here is what I found in jstack

```
"pool-8-thread-10" #92 prio=5 os_prio=0 tid=0x00007f0de4020800 nid=0x48fd runnable [0x00007f0da9ffb000]
   java.lang.Thread.State: RUNNABLE
	at scala.util.hashing.MurmurHash3.listHash(MurmurHash3.scala:169)
	at scala.util.hashing.MurmurHash3$.seqHash(MurmurHash3.scala:221)
	at scala.collection.LinearSeqLike.hashCode(LinearSeqLike.scala:37)
	at scala.collection.LinearSeqLike.hashCode$(LinearSeqLike.scala:37)
	at scala.collection.immutable.List.hashCode(List.scala:85)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at coursier.core.Project.hashCode(Definitions.scala:211)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at scala.util.hashing.MurmurHash3.$anonfun$orderedHash$1(MurmurHash3.scala:110)
	at scala.util.hashing.MurmurHash3.$anonfun$orderedHash$1$adapted(MurmurHash3.scala:109)
	at scala.util.hashing.MurmurHash3$$Lambda$390/645476639.apply(Unknown Source)
	at scala.collection.Iterator.foreach(Iterator.scala:937)
	at scala.collection.Iterator.foreach$(Iterator.scala:937)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
	at scala.collection.IterableLike.foreach(IterableLike.scala:70)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.util.hashing.MurmurHash3.orderedHash(MurmurHash3.scala:109)
	at scala.util.hashing.MurmurHash3$.seqHash(MurmurHash3.scala:222)
	at scala.collection.IndexedSeqLike.hashCode(IndexedSeqLike.scala:40)
	at scala.collection.IndexedSeqLike.hashCode$(IndexedSeqLike.scala:40)
	at scala.collection.immutable.Vector.hashCode(Vector.scala:64)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
	at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
	at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:145)
	at lmcoursier.internal.InterProjectRepository.hashCode(InterProjectRepository.scala:7)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
	at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
	at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:145)
	at scala.Tuple2.hashCode(Tuple2.scala:19)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
	at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
	at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:145)
	at scala.Tuple2.hashCode(Tuple2.scala:19)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at scala.util.hashing.MurmurHash3.$anonfun$unorderedHash$1(MurmurHash3.scala:92)
	at scala.util.hashing.MurmurHash3.$anonfun$unorderedHash$1$adapted(MurmurHash3.scala:91)
	at scala.util.hashing.MurmurHash3$$Lambda$723/1434932926.apply(Unknown Source)
	at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:230)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:461)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:461)
	at scala.util.hashing.MurmurHash3.unorderedHash(MurmurHash3.scala:91)
	at scala.util.hashing.MurmurHash3$.mapHash(MurmurHash3.scala:225)
	at scala.collection.GenMapLike.hashCode(GenMapLike.scala:35)
	at scala.collection.GenMapLike.hashCode$(GenMapLike.scala:35)
	at scala.collection.AbstractMap.hashCode(Map.scala:59)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at scala.util.hashing.MurmurHash3.productHash(MurmurHash3.scala:64)
	at scala.util.hashing.MurmurHash3$.productHash(MurmurHash3.scala:211)
	at scala.runtime.ScalaRunTime$._hashCode(ScalaRunTime.scala:145)
	at coursier.core.Resolution.hashCode(Resolution.scala:626)
	at scala.runtime.Statics.anyHash(Statics.java:110)
	at coursier.graph.DependencyTree$Node.hashCode(DependencyTree.scala:43)
	at scala.collection.mutable.FlatHashTable.addEntry(FlatHashTable.scala:149)
	at scala.collection.mutable.FlatHashTable.addEntry$(FlatHashTable.scala:148)
	at scala.collection.mutable.HashSet.addEntry(HashSet.scala:39)
	at scala.collection.mutable.FlatHashTable.addElem(FlatHashTable.scala:140)
	at scala.collection.mutable.FlatHashTable.addElem$(FlatHashTable.scala:139)
	at scala.collection.mutable.HashSet.addElem(HashSet.scala:39)
	at scala.collection.mutable.HashSet.$plus$eq(HashSet.scala:58)
	at coursier.graph.ReverseModuleTree$.fromDependencyTree(ReverseModuleTree.scala:111)
	at coursier.graph.ReverseModuleTree$.apply(ReverseModuleTree.scala:137)
	at coursier.graph.Conflict$.conflicted(Conflict.scala:34)
	at coursier.graph.Conflict$.apply(Conflict.scala:68)
	at lmcoursier.internal.SbtUpdateReport$.$anonfun$apply$1(SbtUpdateReport.scala:269)
	at lmcoursier.internal.SbtUpdateReport$$$Lambda$6395/2017092449.apply(Unknown Source)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
	at scala.collection.TraversableLike$$Lambda$6/1439337960.apply(Unknown Source)
	at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:230)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:461)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:461)
	at scala.collection.TraversableLike.map(TraversableLike.scala:233)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:226)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at lmcoursier.internal.SbtUpdateReport$.apply(SbtUpdateReport.scala:231)
	at lmcoursier.internal.UpdateRun$.update(UpdateRun.scala:94)
	- locked <0x00000006ccd4df90> (a java.lang.Object)
	at coursier.sbtcoursier.UpdateTasks$.$anonfun$updateTask$19(UpdateTasks.scala:138)
	at coursier.sbtcoursier.UpdateTasks$$$Lambda$6272/2118554574.apply(Unknown Source)
	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
	at scala.Function1$$Lambda$609/2144282958.apply(Unknown Source)
	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:40)
	at sbt.internal.util.$tilde$greater$$Lambda$1891/431446827.apply(Unknown Source)
	at sbt.std.Transform$$anon$4.work(System.scala:67)
	at sbt.Execute.$anonfun$submit$2(Execute.scala:269)
	at sbt.Execute$$Lambda$1911/1684278039.apply(Unknown Source)
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
	at sbt.Execute.work(Execute.scala:278)
	at sbt.Execute.$anonfun$submit$1(Execute.scala:269)
	at sbt.Execute$$Lambda$1899/889346570.apply(Unknown Source)
	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
	at sbt.ConcurrentRestrictions$$anon$4$$Lambda$1906/11646065.apply(Unknown Source)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by alexarchambault** [comment]
@jilen That's https://github.com/coursier/coursier/issues/1251, fixes in the `2.0.0-RC2-1` version of the plugin.

**Comment by jilen** [comment]
Yes, Exactly!!
